client/pages/edit-event.jsx

```diff
@@ -0,0 +1,172 @@
+  }
+
+  handleUpdate(event) {
+    event.preventDefault();
+    const geolocation = `${this.state.address}, +${this.state.city}, +${this.state.state} `;
+    fetch(`https://maps.googleapis.com/maps/api/geocode/json?address=${geolocation}&key=AIzaSyATRROv2KEQF0wX2e5OPR1CCbNaWFgrpcA`)
+      .then(response => response.json()).then(data => data.results[0].geometry.location).then(geoLatlon => {
+        this.setState({ mapLocation: geoLatlon });
+        fetch(`/api/events/${this.props.eventID}`, {
+          method: 'PATCH',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({ ...this.state, ...geoLatlon })
+        });
+      }
+      )
+      .then(() => { location.hash = '#'; });
+  }
+
+  render() {
+    return (
+      <>
+        <div className='form-container'>
+          <form onSubmit={this.handleUpdate}>
+            <div className='form-inputs'>
+              <label>Event Title</label>
+              <input
+                name="title"
+                type="text"
+                id="eventTitle"
+                defaultValue={this.state.event.title}
+                onChange={this.handleChange} />
+            </div>
+            <div className='form-inputs'>
+              <label>Pick a Date</label>
+              <DatePicker selected={this.state.event.date ? new Date(this.state.event.date) : null}
+                onChange={date => this.handleDate(date)}
+              />
+            </div>
+            <div className='form-inputs'>
+              <label>Starting Time</label>
+              <input
+                name="startingtime"
+                type="time"
+                id="startingTime"
+                defaultValue={this.state.event.startingtime}
+                onChange={this.handleChange} />
+            </div>
+            <div className='form-inputs'>
+              <label>Pick a Place</label>
+              <input
+                name="address"
+                type="text"
+                id="address"
+                defaultValue={this.state.event.address}
+                onChange={this.handleAddress} />
+            </div>
+            <div className='form-inputs'>
+              <label>City</label>
+              <input
+                name="city"
+                type="text"
+                id="city"
+                defaultValue={this.state.event.city}
+                onChange={this.handleCity} />
+            </div>
+            <div className='form-inputs'>
+              <label>State</label>
+              <input
+                name="state"
+                type="text"
+                id="state"
+                defaultValue={this.state.event.state}
+                onChange={this.handleState} />
+            </div>
+            <div className='submit-btn'>
+              <button onClick={this.modalOpen}>Delete</button>
+              <button type="submit">Edit Event</button>
+            </div>
```

Comment:
For usability, I'd highly recommend including a "Cancel Changes" button that goes back to the event's details without saving any changes.

---

![image](https://user-images.githubusercontent.com/16143771/148587892-280aa9f9-8d26-4080-84b1-384673e6f76e.png)